### DISCUSSION TITLE
Proper Documentation

### DISCUSSION THREAD

**Discussion by smitsavaliya-1258** [Tinker-Twins/AutoDRIVE#2] [Q&A] [answered]
First i appreciate to make this kind of simulator for project purpose, which helps us a lot by making it open source. But there is only one problem which is proper documentations not available. and yours python and ROS api is not working. i have tried all the ways still it connect to simulator but didn't work. if you can't tell me here then reply me in my mail smitsavaliya2001@gmail.com

I need your help for making my project successful and complete my graduation project.

**Comment by Tinker-Twins** [comment]
Thanks a lot for your appreciation for AutoDRIVE! This project is very close to our heart for several reasons and we will keep trying our best to make sure that a lot of people benifit from it.

Regarding documentation, we have currently included instructions that can help people (having some experience working with similar platforms/ecosystems) get started easily. That being said, we are currently working on creating a comprehensive documentation for all the components of AutoDRIVE ecosystem, releasing the technical report and even writing a paper on it. Please understand that this may take some time since it is just the two of us working on this (among other things), but it will defenitely be available very soon!

Regarding the APIs, we have extensively verified that the C++ and Python APIs as well as the ROS package are all working as expected in single as well as distributed computing settings. Could you please share the exact problem/error that you are encountering?  We can also help you over a video call (over a weekend) if you are not able to get going.

**Comment by smitsavaliya-1258** [comment]
Hi, Thanks for replying

We can connect anytime whenever you guys are available. I am from India so
do mention time and date when you are available.

My system specs are :-
Ubuntu 20.04
Ros Noetic full
Python 3.7 and 3.8 both

Regarding you both is there anything i can help in documentation or your work for making simulator better or anything else do let me know, happy to help you both.

**Comment by Tinker-Twins** [comment, marked as answer]
Hope today's discussion was helpful!

Following is a quick review of today's discussion:

- [AutoDRIVE Simulator Paper](https://arxiv.org/pdf/2103.10030.pdf) hosts details that can serve as potential documentation in addition to the various README files in this repository.
- [AutoDRIVE Devkit's Python API](https://github.com/Tinker-Twins/AutoDRIVE/tree/AutoDRIVE-Devkit/ADSS%20Toolkit/autodrive_py) and/or [AutoDRIVE Devkit's ROS Package](https://github.com/Tinker-Twins/AutoDRIVE/tree/AutoDRIVE-Devkit/ADSS%20Toolkit/autodrive_ros) has the following dependencies (tested with Python 3.6.5):

    - Websocket-related dependencies for communication bridge between [AutoDRIVE Simulator](https://github.com/Tinker-Twins/AutoDRIVE/tree/AutoDRIVE-Simulator) and [AutoDRIVE Devkit](https://github.com/Tinker-Twins/AutoDRIVE/tree/AutoDRIVE-Devkit) (version sensitive):

        | Package | Tested Version |
        |:---------:|:-----------------:|
        | eventlet | 0.33.3 |
        | Flask | 1.1.1 |
        | Flask-SocketIO | 4.1.0 |
        | python-socketio | 4.2.0 |
        | python-engineio | 3.13.0 |
        | greenlet | 1.0.0 |
        | gevent | 21.1.2 |
        | gevent-websocket | 0.10.1 |

        ```bash
        $ pip3 install eventlet==0.33.3
        $ pip3 install Flask==1.1.1
        $ pip3 install Flask-SocketIO==4.1.0
        $ pip3 install python-socketio==4.2.0
        $ pip3 install python-engineio==3.13.0
        $ pip3 install greenlet==1.0.0
        $ pip3 install gevent==21.1.2
        $ pip3 install gevent-websocket==0.10.1
        ```

    - Generic dependencies for data processing and visualization (usually any version will do the job):

        | Package | Tested Version |
        |:---------:|:-----------------:|
        | attrdict | 2.0.1 |
        | numpy | 1.13.3 |
        | pillow | 5.1.0 |
        | opencv-contrib-python | 4.5.1.48 |

        ```bash
        $ pip3 install attrdict
        $ pip3 install numpy
        $ pip3 install pillow
        $ pip3 install opencv-contrib-python
        ```

**Reply by s-s-1997** [reply]
Thanks, this was very helpful!

**Comment by smitsavaliya-1258** [comment]
Thanks, for helping and solving the issue for me.
